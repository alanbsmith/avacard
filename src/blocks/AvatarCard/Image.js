import styled from 'styled-components';
import PropTypes from 'prop-types';

import { dimensions } from '../../utils';

const Image = styled.img.attrs({
  src: props => `https://api.adorable.io/avatars/150/${props.email}.png`,
  alt: 'adorable user avatar',
})`
  border-radius: ${dimensions('borderRadius')};
  height: 106px;
  width: 106px;
`;

Image.propTypes = {
  email: PropTypes.string.isRequired,
};

export default Image;
