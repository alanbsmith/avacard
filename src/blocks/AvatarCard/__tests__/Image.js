import { generateSnapshots } from "__tests__/helpers";

import Image from '../Image';

generateSnapshots(Image, { email: 'user@example.com' });
