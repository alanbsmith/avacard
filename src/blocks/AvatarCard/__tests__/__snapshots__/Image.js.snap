// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styled.img renders correctly 1`] = `
.c0 {
  border-radius: 3px;
  height: 106px;
  width: 106px;
}

<img
  alt="adorable user avatar"
  className="c0"
  src="https://api.adorable.io/avatars/150/user@example.com.png"
/>
`;
